homeassistant/components/rejseplanen/sensor.py

```diff
@@ -203,29 +206,42 @@ def intersection(lst1, lst2):
         for item in results:
             route = item.get("name")
 
-            due_at_date = item.get("rtDate")
-            due_at_time = item.get("rtTime")
+            scheduled_date = item.get("date")
+            scheduled_time = item.get("time")
+            real_time_date = due_at_date = item.get("rtDate")
+            real_time_time = due_at_time = item.get("rtTime")
 
             if due_at_date is None:
-                due_at_date = item.get("date")  # Scheduled date
+                due_at_date = scheduled_date
             if due_at_time is None:
-                due_at_time = item.get("time")  # Scheduled time
+                due_at_time = scheduled_time
 
             if (
                 due_at_date is not None
                 and due_at_time is not None
                 and route is not None
             ):
                 due_at = f"{due_at_date} {due_at_time}"
+                scheduled_at = f"{scheduled_date} {scheduled_time}"
 
                 departure_data = {
+                    ATTR_DIRECTION: item.get("direction"),
                     ATTR_DUE_IN: due_in_minutes(due_at),
                     ATTR_DUE_AT: due_at,
-                    ATTR_TYPE: item.get("type"),
+                    ATTR_FINAL_STOP: item.get("finalStop"),
                     ATTR_ROUTE: route,
-                    ATTR_DIRECTION: item.get("direction"),
+                    ATTR_SCHEDULED_AT: scheduled_at,
```

Comment:
Scheduled at should preferably be a separate timestamp sensor that returns absolute UTC isoformatted time.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
All timestamp times should preferably be this device class sensor instead of an attribute.

Reply:
I'm not sure how much it makes sense to move it to a separate entity. Especially when the departures in `next_departures` also have it, unless I'm then supposed to create two new entities for every single departure. It would make sense to convert it to ISO format though.

Reply:
Although, either way, it would be a breaking change.

Reply:
Actually I'd rather change the state of the sensor to the due_at datetime (which is always the most up to date departure time). If I understand it correctly, the frontend will automatically convert that to a relative time, correct?

Reply:
In Lovelace you can use an Entities card to get a relative time. It's not the default yet.

https://www.home-assistant.io/lovelace/entities/#format

Reply:
I think we should wait with making breaking changes until someone wants to reshape the integration more. I'm ok with adding these features for now as is.

Reply:
Yeah, that sounds like a good plan.